proxy_agent/src/proxy/proxy_authentication.rs

```diff
@@ -313,9 +315,23 @@ pub fn get_authenticate(ip: String, port: u16, claims: Claims) -> Box<dyn Authen
     }
 }
 
+pub fn authenticate(
+    ip: String,
+    port: u16,
+    connection_id: u128,
+    request_uri: hyper::Uri,
+    claims: Claims,
+    shared_state: Arc<Mutex<SharedState>>,
+) -> bool {
+    let auth = get_authenticate(ip, port, claims);
+    Connection::write(connection_id, format!("Got auth: {}", auth.to_string()));
```

Comment:
Is there a better name for the enum than Authenticate? It would improve clarity, but also be easier to track safety. When I first read this line I was thinking we were sending a credential somewhere

Reply:
rename it to Authorizer and relative mod, trait and fn names.